cognite/client/_api/files.py

```diff
@@ -612,19 +612,52 @@ def retrieve_download_urls(
         results = tasks_summary.joined_results(unwrap_fn=lambda res: res.json()["items"])
         return {result.get("id") or result["externalId"]: result["downloadUrl"] for result in results}
 
+    @staticmethod
+    def _create_unique_file_names(file_names_in: list[str] | list[Path]) -> list[str]:
+        """
+        Create unique file names by appending a number to the base file name.
+        """
+        file_names: list[str] = [str(file_name) for file_name in file_names_in]
+        file_names_unique_original = set(file_names)
+        file_names_unique_created: list = []
+        file_names_original_count: dict = {}
```

Comment:
Although _explicit is better than implicit_, with this much overlap between variable names, the overall readability of the function goes down (in my opinion 😉 ). I think that in a function dealing with file names, it would be fine to use variables like `count`, `original` and `newly_created` or perhaps just `changed`.